win32_deps_build.sh

```diff
@@ -42,7 +42,51 @@ dokanTag="v1.3.1.1000"
 dokanSrcDir="${depsSrcDir}/dokany"
 dokanLibDir="${depsToolsetDir}/dokany/lib"
 
+# Allow for OS specific customizations through the OS flag (normally
+# passed through from win32_build).
+# Valid options are currently "ubuntu" and "suse".
+OS=${OS:-"ubuntu"}
+
+# MINGW Settings:
+# Due to inconsistencies between distributions, mingw versions, binaries,
+# and directories must be determined (or defined) first.
+# -Common mingw settings-
 MINGW_PREFIX="x86_64-w64-mingw32-"
+MINGW_BASE="x86_64-w64-mingw32"
+MINGW_CPP="${MINGW_BASE}-c++"
+MINGW_DLLTOOL="${MINGW_BASE}-dlltool"
+MINGW_WINDRES="${MINGW_BASE}-windres"
+MINGW_STRIP="${MINGW_BASE}-strip"
+# -Distribution specific mingw settings-
+case "$OS" in
+    ubuntu)
+        mingwPosix="-posix"
+        mingwLibDir="/usr/lib/gcc"
+        mingwVersion="$(${MINGW_CPP}${mingwPosix} -dumpversion)"
+        mingwTargetLibDir="${mingwLibDir}/${MINGW_BASE}/${mingwVersion}"
+        mingwLibpthreadDir="/usr/${MINGW_BASE}/lib"
+        PTW32Include=/usr/share/mingw-w64/include
+        PTW32Lib=/usr/x86_64-w64-mingw32/lib
+       ;;
+    suse)
+        mingwPosix=""
+        mingwLibDir="/usr/lib64/gcc"
+        mingwVersion="$(${MINGW_CPP}${mingwPosix} -dumpversion)"
+        mingwTargetLibDir="/usr/${MINGW_BASE}/sys-root/mingw/bin"
+        mingwLibpthreadDir="$mingwTargetLibDir"
+        PTW32Include=/usr/x86_64-w64-mingw32/sys-root/mingw/include
+        PTW32Lib=/usr/x86_64-w64-mingw32/sys-root/mingw/lib
+        ;;
+    *)
+        echo "$ID is unknown, automatic configuration is not possible."
+        exit 1
+        ;;
+esac
+# -Common mingw settings, dependent upon distribution specific settings-
+MINGW_FIND_ROOT_LIB_PATH="${mingwLibDir}/\${TOOLCHAIN_PREFIX}/${mingwVersion}"
+MINGW_CC="${MINGW_BASE}-gcc${mingwPosix}"
+MINGW_CXX="${MINGW_BASE}-g++${mingwPosix}"
+# End MINGW configuration
```

Comment:
We could move all of this to a separate file (e.g. mingw_conf.sh) and source it in both scripts. I'll check if the same toolchain file can be re-used for ceph as well as its dependencies. The "_POSIX" flags might not work with some of the dependencies.

Anyway, we can clean it up afterwards.

Reply:
Yes, that would be a much better approach. Let me know if you need any validations on the SUSE side.